scripts/nodelib.py

```diff
@@ -48,15 +48,16 @@ def nodocker_init(home_dir, is_release, init_flags):
     subprocess.call(cmd + init_flags)
 
 
-"""Retrieve requested chain id from the flags."""
-
-
 def get_chain_id_from_flags(flags):
-    # TODO this doesn't handle the `--chain-id my-id` syntax which should also be valid
-    chain_id_flags = [flag for flag in flags if flag.startswith('--chain-id=')]
-    if len(chain_id_flags) == 1:
-        return chain_id_flags[0][len('--chain-id='):]
-    return ''
+    """Retrieve requested chain id from the flags."""
+    chain_id = None
+    flags_iter = iter(flags)
+    for flag in flags_iter:
+        if flag.startswith('--chain-id='):
```

Comment:
I'd like to avoid the magic number 11. For example, it's not obvious whether 11 includes the equal sign or not.